### PR TITLE
Adding CI to PlayerKit

### PR DESCRIPTION
**Pull request by MathiasGr**
Here's the setup for CI @ghking.

### CODE REVIEW

**Comment by ghking** [comment]
@MathiasGr Dope thank you!

File: fastlane/Fastfile [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+# This is the minimum version number required.
+# Update this, if you use features of a newer version
+fastlane_version "1.103.0"
+
+default_platform :ios
+
+platform :ios do
+  before_all do
+    ensure_xcode_version(version: "8.0")
```
Comment (wilrnh) [inline fastlane/Fastfile]: Just bump this up to 8.2.1

File: circle.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+general:
+  artifacts:
+    - "xcodebuild.log"
+    - "build/Logs"
+machine:
+  xcode:
+    version: 8.0
```
Comment (wilrnh) [inline circle.yml]: Just bump this to 8.2.1

File: circle.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+dependencies:
+    override:
+        - bundle install
+    cache_directories:
+        - "~/.cocoapods"
+        - "Pods"
```
Comment (wilrnh) [inline circle.yml]: Is this really the path to the Pods folder, maybe its actually `Example/Pods`?

File: circle.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+        - "vendor/bundle"
```
Comment (wilrnh) [inline circle.yml]: `bundle install` doesnt create `vendor/bundle`; use `bundle install --deployment` for that

File: fastlane/Fastfile [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+  end
+
+  desc "build the example project"
+  lane :build_example do |options|
```
Comment (wilrnh) [inline fastlane/Fastfile]: no need for build_example, :test already runs "build" and "test"

File: .gitignore [hunk 6]
```diff
@@ -49,14 +49,13 @@ Pods
...
-# https://docs.fastlane.tools/best-practices/source-control/#source-control
 
+# https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Gitignore.md
 fastlane/report.xml
 fastlane/Preview.html
 fastlane/screenshots
 fastlane/test_output
+
+# Exclude fastlane xcode log
+xcodebuild.log
```
Comment (wilrnh) [inline .gitignore]: only build_example creates xcodebuild.log, see circle.yml

File: circle.yml [hunk 7]
```diff
@@ -0,0 +1,19 @@
+general:
+  artifacts:
+    - "xcodebuild.log"
```
Comment (wilrnh) [inline circle.yml]: nope, see .gitignore

File: circle.yml [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+    - "build/Logs"
```
Comment (wilrnh) [inline circle.yml]: build/Logs are the xcactivity logs, we'd also need to save the `PlayerKit_Example-PlayerKit-Example.log` here too

File: fastlane/Fastfile [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+      configuration: "Debug",
+      buildlog_path: ".",
+      derived_data_path: "build/",
+      formatter: 'xcpretty-json-formatter'
+    )
+
+    pretty_junit(file_pattern: "fastlane/test_output/report.junit")
+
+    if is_ci?
+      danger(use_bundle_exec: false)
```
Comment (wilrnh) [inline fastlane/Fastfile:42]: :o Gotta get a Dangerfile for this to work...

File: fastlane/Fastfile [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+  lane :test do |options|
+
+    cocoapods(
+      podfile: "Example/"
+    )
+
+    scan(
+      workspace: "Example/PlayerKit.xcworkspace",
+      scheme: "PlayerKit-Example",
+      configuration: "Debug",
```
Comment (wilrnh) [inline fastlane/Fastfile:33]: Sure you want to test Debug not Release?

**Review by wilrnh — CHANGES_REQUESTED** [review]

File: fastlane/Fastfile (hunk 9 above)
Comment (MathiasGr) [reply fastlane/Fastfile:42]: done

File: fastlane/Fastfile (hunk 10 above)
Comment (MathiasGr) [reply fastlane/Fastfile:33]: Not sure about this one. @ghking do you have any idea?

**Comment by MathiasGr** [comment]
@wilrnh Addressed issues

**Comment by ghking** [comment]
@wilrnh @MathiasGr bump :)

File: fastlane/Fastfile (hunk 10 above)
Comment (wilrnh) [reply fastlane/Fastfile:33]: @MathiasGr lets switch this to Release. I expect when this is a cocoapod we'll want it to build with release settings.

**Review by wilrnh — APPROVED** [review]
🔥

**Comment by ghking** [comment]
@wilrnh @MathiasGr Guys can we merge this?

**Comment by wilrnh** [comment]
😱

**Pull request merged**